src/components/Admin/Admin.js

```diff
@@ -1,53 +1,164 @@
 import React, {Component} from 'react';
+import PropTypes from 'prop-types';
 import initFirebase from '../../utils/firebase';
 import firebase from 'firebase/app';
 import 'firebase/auth';
 import './Admin.css';
 
 class Admin extends Component {
   state = {
-    authed: 'Loading...'
+    status: 'Loading...',
+    loggedIn: false,
+    admin: false,
+    models: [],
+    currentModel: '',
+    modelSelect: ''
   };
 
-  componentDidMount() {
-    initFirebase();
+  authenticate = () => {
     firebase
       .auth()
       .getRedirectResult()
       .then(result => {
         if (result.user) {
-          this.setState({
-            authed: 'Welcome, ' + result.user.displayName
+          const userRef = firebase
+            .firestore()
+            .collection('users')
+            .doc(result.user.uid);
+
+          userRef.get().then(user => {
+            this.setState({loggedIn: true});
+            if (user.exists) {
+              if (user.data().admin) {
+                this.initAdmin(user.data().name);
+              } else {
+                userRef.set({
+                  name: result.user.displayName,
+                  admin: false
+                });
+                this.setState({status: 'You are not an administrator.'});
+              }
+            } else {
+              this.setState({status: 'You are not an administrator.'});
+            }
           });
         } else {
-          this.setState({authed: 'You are not authenticated.'});
+          this.setState({status: 'You are not authenticated.'});
         }
+      })
+      .catch(function(error) {
+        this.setState({status: error});
+      });
+  };
+
+  login = () => {
+    firebase.auth().signInWithRedirect(new firebase.auth.GoogleAuthProvider());
+  };
+
+  logout = () => {
+    firebase
+      .auth()
+      .signOut()
+      .then(() => {
+        this.setState({loggedIn: false, admin: false});
+      });
+  };
+
+  initAdmin = name => {
+    this.setState({
+      admin: true,
+      status: 'Welcome, ' + name
```

Comment:
I dont think you should hold full msg strings in state, it would be cleaner to have status:  as an object with 2 properties, One which describes whether the user is an administrator/authenticated and one with the name (can be optional) This way the state doesnt care how you are using the information.

Reply:
As discussed, we agreed the logic needed to make this change would add more complexity than necessary. Since the status variable is purely visual feedback for the user and nothing else, we decided to leave this as is.